### PR TITLE
Implement Backend/Frontend connection

### PR DESCRIPTION
**Pull request by MKRoughDiamond**
### frontend spec
```
{
  "totalSimData" : [
    {
      "major" : {
        "name" : string,
        "type_" : string
      },
      "semesters" : [
        {
          "year" : string,
          "season" : string,
          "subjects" : [
            {
              "title" : string,
              "category" : string,
              "tooltip" : string
            },
            ...
          ]
        },
        ...
      ],
      "remainSubject" : [
        {
          "title" : string,
          "category" : string,
          "tooltip" : string
        },
        ...
      ],
      "creditResults" : {
        "totalReq" : int,
        "totalAcq" : int,
        "mandatoryReq" : int,
        "mandatoryAcq" : int,
        "electivesReq" : int,
        "electivesAcq" : int,
        "liberalReq" : int,
        "liberalAcq" : int
      } 
    },
    ...
  ]
}
```

* total : full of credits
* mandatory : `전필`
* electives : `전선`
* liberal : `교양`, 모든 교양을 포함함(`free` 제거)
* major->type_ : "주전공(단일)", "주전공(복/부)", "부전공", "복수전공"

### CODE REVIEW

**Comment by MKRoughDiamond** [comment]
now it works!

File: frontend/src/elm/Main/Response.elm [hunk 1]
```diff
@@ -0,0 +1,62 @@
+module Main.Response exposing (..)
+
+import Json.Decode as Json
+import Main.Models as Main
+import Utils.Major as Utils
```
Comment (sonowz) [inline frontend/src/elm/Main/Response.elm]: Importing name is weird. I think it should be
`import Main.Models as Models`
`import Utils.Major as Major`

File: frontend/src/elm/Main/Response.elm [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  Json.map8
+    Main.CreditResults
+    ( Json.field "total_req" Json.int )
+    ( Json.field "total_acq" Json.int )
+    ( Json.field "required_req" Json.int )
+    ( Json.field "required_acq" Json.int )
+    ( Json.field "elective_req" Json.int )
+    ( Json.field "elective_acq" Json.int )
+    ( Json.field "liberal_req" Json.int )
+    ( Json.field "liberal_acq" Json.int )
```
Comment (sonowz) [inline frontend/src/elm/Main/Response.elm]: Would be better if it has newline at the end of file.

File: frontend/src/elm/Main/Update.elm [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+import Navigation exposing (load)
+import Http
+import Cmd.Extra
+import GlobalMsgs exposing (GlobalMsg(..))
 import Array
 import Maybe
 import Main.Models exposing (..)
+import Main.Response as Response
 import Main.Msgs exposing (Msg(..))
+import Debug
```
Comment (sonowz) [inline frontend/src/elm/Main/Update.elm]: I think importing orders can be rearranged here.

File: frontend/src/elm/Utils/Major.elm [hunk 4]
```diff
@@ -26,6 +28,27 @@ typeToString majorType =
...
+      Just DoubleMajor
+    _ ->
+      Nothing
+
+
+normalization : Maybe MajorType -> MajorType
+normalization type_ = Maybe.withDefault MajorSingle type_
+
+stringToTypeNormal : String -> MajorType
+stringToTypeNormal input = normalization (stringToType input)
```
Comment (sonowz) [inline frontend/src/elm/Utils/Major.elm]: What does `Normal` mean? I think it should be more specific.

**Review by sonowz — CHANGES_REQUESTED** [review]

File: frontend/src/elm/Main/Response.elm (hunk 2 above)
Comment (Jhuni0123) [reply frontend/src/elm/Main/Response.elm]: We can use `editorconfig` to force inserting newline at the end of file.

File: frontend/src/elm/Main.elm [hunk 5]
```diff
@@ -15,8 +16,15 @@ init location =
     currentRoute =
       Routes.parseLocation location
 
+    commands : Routes.Route -> (Cmd Msg)
```
Comment (Jhuni0123) [inline frontend/src/elm/Main.elm]: `initialCommand` is better than `commands`

**Comment by Jhuni0123** [comment]
I think small commits that fix minor things should be squashed.

File: frontend/src/elm/Main/Response.elm (hunk 2 above)
Comment (MKRoughDiamond) [reply frontend/src/elm/Main/Response.elm]: fixed

**Comment by skystar-p** [comment]
Is this PR done?

**Review by sonowz — APPROVED** [review]
I'll approve this.

**Comment by MKRoughDiamond** [comment]
done

**Pull request merged**